<launch>
    <!-- determines whether the fake or real arm is being controlled -->
    <arg name="fake_mode" default="true"/>
    <!-- coordinates to determine where the arm reaches to pick up an object -->
    <!-- by default, it reaches right in front of it -->
    <arg name="x" default="0.5"/>
    <arg name="y" default="0.0"/>
    <arg name="z" default="0.05"/>


    <!-- if fake_mode is being launched -->
    <group if="$(var fake_mode)">
        <include file="$(find-pkg-share franka_moveit_config)/launch/moveit.launch.py">
            <arg name="robot_ip" value="dont-care" />
            <arg name="use_fake_hardware" value="true" />
        </include>

        <node 
            name="picker" 
            pkg="pickmeup"
            exec="picker">
            <param name="goal_x" value="$(var x)" />
            <param name="goal_y" value="$(var y)" />
            <param name="goal_z" value="$(var z)" />
        </node>
    </group>

    <!-- if the real arm is being launched -->
    <group unless="$(var fake_mode)">
        <include file="$(find-pkg-share franka_moveit_config)/launch/moveit.launch.py">
            <arg name="robot_ip" value="panda0.robot" />
        </include>

        <node 
            name="picker" 
            pkg="pickmeup"
            exec="picker">
            <param name="goal_x" value="0.5" />
            <param name="goal_y" value="0.0" />
            <param name="goal_z" value="0.05" />
            <param name="fake_mode" value="false" />
        </node>
    </group>

</launch>